File: insertionSortLuis.js
let vetor = [20, 8, 99, 800, 9, 1]
let atual = []
let newValue = window.prompt('insira um valor')

vetor.push(Number(newValue))

console.log(vetor);

for (let prox = 1; prox < vetor.length; prox++) {
  let ant = prox - 1
  atual = vetor[prox]
  for (ant >= 0; atual < vetor[ant]; ant--) {
    vetor[ant + 1] = vetor[ant]
  }
  vetor[ant + 1] = atual
}

console.log(vetor)

File: insertionSortLuis.py

import random
import timeit

x = 100
vetor = [20, 8, 99, 800, 9, 1, 801]
vetor = random.sample(range(5000000000000000), x)

atual = []
passos = 0
inicio = timeit.default_timer()

for prox in range(1, len(vetor)):
    # print(vetor[prox])
    ant = prox - 1
    atual = vetor[prox]
    while ant >= 0 and atual < vetor[ant]:
        passos = passos + 1
        vetor[ant+1] = vetor[ant]
        ant -= 1
    vetor[ant + 1] = atual
fim = timeit.default_timer()
print('tempo...: %f ' % (fim - inicio))
print('número de passos: %d ' % passos)

print(vetor)
